crates/amaru/src/bin/amaru/cmd/daemon.rs

```diff
@@ -75,11 +75,11 @@ pub struct Args {
 
 pub async fn run(
     args: Args,
-    metrics: Option<SdkMeterProvider>,
+    provider: Option<SdkMeterProvider>,
 ) -> Result<(), Box<dyn std::error::Error>> {
     let config = parse_args(args)?;
 
-    let metrics = metrics.map(track_system_metrics).transpose()?;
+    let metrics = provider.clone().map(track_system_metrics).transpose()?;
```

Comment:
🤨

Reply:
I need to pass the meter provider to the initialization of the stages which occurs later, so I have to clone it here.